From 1d73990e7d433456a02bfb4b283b813bf2385c35 Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Tue, 13 Apr 2021 11:26:22 +0300
Subject: [PATCH] [SW-31823] boot_if: cpu_boot_dev_sts_get_all

The proto_status FWLP command, needs to have access to all
the device's CAPS status. Till now the access is for writing,
or simple bits operations.

Change-Id: Ief561068b35be7ad8efe08a6d28f6ba2435ebd9f
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/common/boot_if.c b/common/boot_if.c
index 84a6962..00134d0 100644
--- a/common/boot_if.c
+++ b/common/boot_if.c
@@ -68,3 +68,12 @@
 {
 	regmap_writel(sts->val, mmCPU_CMD_STATUS_TO_HOST);
 }
+
+u64 cpu_boot_dev_sts_get_all(void)
+{
+	u64 ret;
+
+	ret = ((u64)regmap_readl(mmCPU_BOOT_DEV_STS1) << 32);
+	ret |= (u64)regmap_readl(mmCPU_BOOT_DEV_STS0);
+	return ret;
+}
diff --git a/include/boot_if.h b/include/boot_if.h
index 5c8b77d..c709295 100644
--- a/include/boot_if.h
+++ b/include/boot_if.h
@@ -16,6 +16,7 @@
 void cpu_boot_err_clrset(u32 mask, bool set);
 void cpu_boot_dev_sts_clrset(u32 mask, bool set);
 void cpu_boot_rsp_set(enum cpu_msg_status sts);
+u64 cpu_boot_dev_sts_get_all(void);
 
 /* COMMS APIs */
 void cpu_comms_sts_set(struct comms_status *sts);
